RevenueCatUI/Templates/Components/Text/TextComponentViewModel.swift

```diff
@@ -17,51 +17,240 @@ import SwiftUI
 
 #if PAYWALL_COMPONENTS
 
+struct LocalizedTextPartial: LocalizedPartial {
+
+    let text: String?
+    let partial: PaywallComponent.PartialTextComponent
+
+}
+
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 class TextComponentViewModel {
 
     private let localizedStrings: PaywallComponent.LocalizationDictionary
     private let component: PaywallComponent.TextComponent
 
-    let text: String
+    private let text: String
+    private let localizedStates: LocalizedStates<LocalizedTextPartial>?
+    private let localizedConditions: LocalizedConditions<LocalizedTextPartial>?
 
     init(localizedStrings: PaywallComponent.LocalizationDictionary, component: PaywallComponent.TextComponent) throws {
         self.localizedStrings = localizedStrings
         self.component = component
         self.text = try localizedStrings.string(key: component.text)
-    }
 
-    var fontFamily: String? {
-        component.fontFamily
+        // Localize state partials
+        self.localizedStates = try self.component.state.flatMap({ state in
+            LocalizedStates(
+                selected: try state.selected.flatMap({ partial in
+                    try LocalizedTextPartial.create(from: partial, using: localizedStrings)
+                }),
+                introOffer: try state.introOffer.flatMap({ partial in
+                    try LocalizedTextPartial.create(from: partial, using: localizedStrings)
+                })
+            )
+        })
+
+        self.localizedConditions = try self.component.conditions.flatMap({ condition in
+            LocalizedConditions(
+                mobileLandscape: try condition.mobileLandscape.flatMap({ partial in
+                    try LocalizedTextPartial.create(from: partial, using: localizedStrings)
+                }),
+                tablet: try condition.tablet.flatMap({ partial in
+                    try LocalizedTextPartial.create(from: partial, using: localizedStrings)
+                }),
+                tabletLandscape: try condition.tabletLandscape.flatMap({ partial in
+                    try LocalizedTextPartial.create(from: partial, using: localizedStrings)
+                }),
+                desktop: try condition.desktop.flatMap({ partial in
+                    try LocalizedTextPartial.create(from: partial, using: localizedStrings)
+                })
+            )
+        })
+
     }
 
-    var fontWeight: Font.Weight {
-        component.fontWeight.fontWeight
+    @ViewBuilder
+    func styles(
+        state: ComponentViewState,
+        condition: ScreenCondition,
+        application: @escaping (TextComponentStyle) -> some View
```

Comment:
Naming suggestion: `apply`. `application` makes me think of a computer program haha, _apps_ as the youth say.

Reply:
Honestly... this was totally meant to be `apply` 😅 I think I had computer programs on my mind 🤷‍♂️